hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/ledger/accounts/HederaAccountCustomizer.java

```diff
@@ -81,7 +81,10 @@ public void customizeSynthetic(final ContractCreateTransactionBody.Builder op) {
             op.setDeclineReward((boolean) changes.get(AccountProperty.DECLINE_REWARD));
         }
         if (changes.containsKey(AccountProperty.AUTO_RENEW_ACCOUNT_ID)) {
-            op.setAutoRenewAccountId(((EntityId) changes.get(AccountProperty.AUTO_RENEW_ACCOUNT_ID)).toGrpcAccountId());
+            final var autoRenewAccountId = (EntityId) changes.get(AccountProperty.AUTO_RENEW_ACCOUNT_ID);
```

Comment:
Is this probably because we don't validate auto_renew_account_id is valid before customizing ?

Reply:
Afaict it is only to have `EntityId.MISSING_ID` passed into this method when:
 - The "parent" contract is a `MerkleAccount` (**not** `OnDiskAccount`) whose `autoRenewId` field is set to `EntityId.MISSING_ID`.

And **that** is only possible when the parent contract is a `MerkleAccount` that:
 - Originally had an `autoRenewId`; and then,
 - Had that cleared to `0.0.0` via a `ContractUpdate`.

Since we plan to move accounts to disk in 0.42, the first `MerkleAccount` condition will no longer be met. So this check is just an extra defense measure, in case there is some other possibility I am not seeing. 🤷